src/IECoreDelight/Renderer.cpp

```diff
@@ -986,8 +986,39 @@ IECore::InternedString g_frameOptionName( "frame" );
 IECore::InternedString g_cameraOptionName( "camera" );
 IECore::InternedString g_sampleMotionOptionName( "sampleMotion" );
 IECore::InternedString g_oversamplingOptionName( "dl:oversampling" );
+
+IECore::InternedString g_maxLengthDiffuseOptionName( "dl:maximumraylength.diffuse" );
+IECore::InternedString g_maxLengthHairOptionName( "dl:maximumraylength.hair" );
+IECore::InternedString g_maxLengthReflectionOptionName( "dl:maximumraylength.reflection" );
+IECore::InternedString g_maxLengthRefractionOptionName( "dl:maximumraylength.refraction" );
+IECore::InternedString g_maxLengthSpecularOptionName( "dl:maximumraylength.specular" );
+IECore::InternedString g_maxLengthVolumeOptionName( "dl:maximumraylength.volume" );
+IECore::InternedString g_clampIndirectOptionName( "dl:clampindirect" );
+IECore::InternedString g_showMultipleScatteringOptionName( "dl:show.multiplescattering" );
+
 const char *g_screenHandle = "ieCoreDelight:defaultScreen";
 
+void setNSIGlobalOption( NSIContext_t context, const InternedString &name, const Object *value )
+{
+	if( value )
+	{
+		if( const Data *data = reportedCast<const Data>( value, "option", name ) )
+		{
+			ParameterList params;
+			params.add( name.c_str() + 3, data );
```

Comment:
I suppose it doesn't _really_ matter, but moving the code with the `+3` into a function here seems error prone, since it's now a long way from the place where we checked the prefix is `dl:`.

Reply:
Fixed in refactor in https://github.com/GafferHQ/gaffer/pull/5333/commits/0ca905e98abe63931686805e3f9a5d1ae018d175